src/runtime/inline/list.h

```diff
@@ -48,12 +48,12 @@ extern "C" inline void listAppendInternalStolen(Box* s, Box* v) {
     assert(PyList_Check(s));
     BoxedList* self = static_cast<BoxedList*>(s);
 
-    assert(self->size <= self->capacity);
     self->ensure(1);
 
-    assert(self->size < self->capacity);
+    assert(self->size < self->allocated);
     self->elts->elts[self->size] = v;
     self->size++;
+    self->allocated = self->size;
```

Comment:
this looks wrong :-( - why did you have to add this?
I think this is will force us to do much more reallocs (after every append)
I checked the perf numbers and it looks like this PR is a quite a perf regression which would be explainable by this change.